### PR TITLE
Model and algorithms refactor

### PR DESCRIPTION
**Pull request by eym55**
I added the gradient clipping with global norms from this paper: https://arxiv.org/pdf/1211.5063.pdf
implemented from here: https://www.tensorflow.org/api_docs/python/tf/clip_by_global_norm
I used a default of 5.0 for the clipping because I saw it here: https://stackoverflow.com/questions/36498127/how-to-apply-gradient-clipping-in-tensorflow
Also did some minor refactoring to add single underscores and making the clip_norm configurable

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -35,14 +27,6 @@
...
     "model_module" : "services.models.sequential.dueling_dqn"
   },
-  "imgobs" : {
-    "environment_name" : "MiniGrid-arie-test-v0",
-    "environment_wrapper" : {
-      "modules" : ["gym_minigrid.wrappers","gym_minigrid.wrappers"], 
-      "classes" : ["RGBImgPartialObsWrapper", "ImgObsWrapper"]
-    },
-    "model_module" : "services.models.sequential.cnn"
-  },
```
Comment (arie-glazier) [inline config.json:45]: dont delete my tests please

File: default_config.json [hunk 2]
```diff
@@ -26,5 +26,6 @@
     "environment_name" : "MiniGrid-arie-test-v0",
     "replay_buffer_module": "services.algorithms.dqn.replay_buffer.simple",
     "replay_buffer_prioritization" : 0.5,
-    "target_sync_frequency" : 50
+    "target_sync_frequency" : 50,
+      "clip_norm": 0
```
Comment (arie-glazier) [inline default_config.json:30]: align with others

File: services/algorithms/__init__.py [hunk 3]
```diff
@@ -14,6 +14,9 @@ def __init__(self, **kwargs):
     self.loss_function = getattr(keras.losses, kwargs[LOSS_FUNCTION])
     self.learning_rate = float(kwargs[LEARNING_RATE])
     self.optimizer = getattr(keras.optimizers, kwargs[OPTIMIZER])(lr=self.learning_rate, epsilon=1.5e-4) #TODO: this epsilon only works for Adam
+    self.clip_norm = float(kwargs[CLIP_NORM])
+    # Does not perform clipping unless clip_norm is set to non-zero float
+    if self.clip_norm == 0: self.clip_norm = None
```
Comment (arie-glazier) [inline services/algorithms/__init__.py:19]: This line isn't necessary for your code to work.  In Python 0.0 == False and 0 == False so your existence check down the line will work either way.

Regardless, let's make the value in default_json "null" and make line 17:
`self.clip_norm = float(kwargs.get(CLIP_NORM)) if kwargs.get(CLIP_NORM) else None`

**Review by arie-glazier — CHANGES_REQUESTED** [review]
Great job, couple of minor/nitpicky comments

File: default_config.json (hunk 2 above)
Comment (eym55) [reply default_config.json:30]: I tried to fix this but it just wouldn't align :/

**Review by arie-glazier — APPROVED** [review]
